client/src/island/hexalot.ts

```diff
@@ -110,15 +115,17 @@ export class Hexalot {
         }
     }
 
-    public fetchJourney(storage: IStorage, island: Island, loaded: () => void): boolean {
+    public fetchJourney(storage: IStorage, island: Island, loaded: (journey: Journey) => void): boolean {
```

Comment:
Make this into a Promise instead, and drop `loaded`.

Reply:
but that would be an async, right?

Reply:
Doesn't have to be. You could just create a promise manually